src/content/en/fundamentals/primers/service-workers/a-basic-service-worker.markdown

```diff
@@ -0,0 +1,170 @@
+The most basic service worker registration looks like this:
+
+{% highlight javascript %}
+// Does the browser support service workers?
+if ('serviceWorker' in navigator) {
+  // Yes
+  navigator.serviceWorker.register('/service-worker.js').then(function() {
+    // Actions not covered by this primer.
+  });
+} else {
+  // service worker not supported.
+};
+{% endhighlight %}
+
+There's more to learn about clients, but that's another lesson. Now let's 
+talk about the service worker itself.
+
+{% include shared/note.liquid list=page.notes.promises %}
+
+## Where can the service worker play?
+
+A client tells the service worker where it can operate. This is called the
+service worker's scope, and it's the highest level of your website where the
+service worker can be used. Let's look at the registration code again, but with
+some changes to the `register()` parameters.
+
+{% highlight javascript %}
+// Does the browser support service workers?
+if ('serviceWorker' in navigator) {
+  // Yes
+  navigator.serviceWorker.register('/service-worker.js', {scope: '/'})
+    .then(function() {
+    // Actions not covered by this primer.
+  });
+} else {
+  // No, but this client should work anyway.
+}
+{% endhighlight %}
+
+This code is functionally identical to the example given in the last section. 
+Both examples specify that the scope of the service worker is the entire domain. 
+If you want to restrict the service worker to part of a site, simply specify it 
+in the scope. 
+
+For example, say that you have an auction site with different features for
+buyers  and sellers. The buyers section is at `example.com/buyers` and the
+sellers section  is at `example.com/sellers`. A service worker with a scope of
+`/buyers` it can only control clients under `example.com/buyers`. It cannot control
+clients under  `example.com/sellers`. Similarly, a service worker with a scope
+of `/sellers` can  only control clients under `example.com/sellers`.
+
+A less than obvious corollary of this that service workers can't be stored in 
+an assets folder. For example a registration like this will always fail:
+
+{% highlight javascript %}
+register('/assets/js/service-worker.js', {scope: '/'})
+{% endhighlight %}
+
+For the scope of this service worker to be `/` it has to be served from 
+`/service-worker.js`
+
+{% include shared/note.liquid list=page.notes.https-only %}
+
+## A service worker installs and activates
+
+Let’s look at what happens during the installation of the service worker. To set 
+up a service worker, you need to implement two event handlers:
+
+* `install`
+* `activate`
+
+These events are fired once for a service worker in a particular scope, no matter 
+how many clients use it.
+
+### Install
+
+Use the install event to get everything this service worker needs to operate.
+This could include initializing caches, or prefetching data.
+
+The install event is kicked off immediately after the script containing it is
+downloaded.
```

Comment:
I'd mention that when there's an update to a service worker script that was previously registered, the `install` event is also fired.